pkg/cache/sql/store/store.go

```diff
@@ -0,0 +1,370 @@
+func (s *Store) Add(obj any) error {
+	key, err := s.keyFunc(obj)
+	if err != nil {
+		return err
+	}
+
+	err = s.upsert(key, obj)
+	return err
+}
+
+// Update saves an obj, or updates it if it exists in this Store
+func (s *Store) Update(obj any) error {
+	return s.Add(obj)
+}
+
+// Delete deletes the given object, if it exists in this Store
+func (s *Store) Delete(obj any) error {
+	key, err := s.keyFunc(obj)
+	if err != nil {
+		return err
+	}
+	return s.deleteByKey(key)
+}
+
+// List returns a list of all the currently known objects
+// Note: I/O errors will panic this function, as the interface signature does not allow returning errors
+func (s *Store) List() []any {
+	rows, err := s.QueryForRows(context.TODO(), s.listStmt)
+	if err != nil {
+		panic(errors.Wrap(err, "Unexpected error in Store.List"))
+	}
+	result, err := s.ReadObjects(rows, s.typ, s.shouldEncrypt)
+	if err != nil {
+		panic(errors.Wrap(err, "Unexpected error in Store.List"))
+	}
+	return result
+}
+
+// ListKeys returns a list of all the keys currently in this Store
+// Note: Atm it doesn't appear returning nil in the case of an error has any detrimental effects. An error is not
+// uncommon enough nor does it appear to necessitate a panic.
+func (s *Store) ListKeys() []string {
+	rows, err := s.QueryForRows(context.TODO(), s.listKeysStmt)
+	if err != nil {
+		fmt.Printf("Unexpected error in store.ListKeys: %v\n", err)
+		return []string{}
+	}
+	result, err := s.ReadStrings(rows)
+	if err != nil {
+		fmt.Printf("Unexpected error in store.ListKeys: %v\n", err)
+		return []string{}
+	}
+	return result
+}
+
+// Get returns the object with the same key as obj
+func (s *Store) Get(obj any) (item any, exists bool, err error) {
+	key, err := s.keyFunc(obj)
+	if err != nil {
+		return nil, false, err
+	}
+
+	return s.GetByKey(key)
+}
+
+// Replace will delete the contents of the Store, using instead the given list
+func (s *Store) Replace(objects []any, _ string) error {
+	objectMap := map[string]any{}
+
+	for _, object := range objects {
+		key, err := s.keyFunc(object)
+		if err != nil {
+			return err
+		}
+		objectMap[key] = object
+	}
+	return s.replaceByKey(objectMap)
+}
+
+// Resync is a no-op and is deprecated
+func (s *Store) Resync() error {
+	return nil
+}
+
+/* Utilities */
+
+// sanitize returns a string  that can be used in SQL as a name
+func sanitize(s string) string {
+	return strings.ReplaceAll(s, "\"", "")
+}
+
+// toBytes encodes an object to a byte slice
+func (s *Store) toBytes(obj any) []byte {
+	var buf bytes.Buffer
+	enc := gob.NewEncoder(&buf)
+	err := enc.Encode(obj)
+	if err != nil {
+		panic(errors.Wrap(err, "Error while gobbing object"))
+	}
+	bb := buf.Bytes()
+	return bb
+}
+
+// fromBytes decodes an object from a byte slice
+func (s *Store) fromBytes(buf sql.RawBytes) (reflect.Value, error) {
+	dec := gob.NewDecoder(bytes.NewReader(buf))
+	singleResult := reflect.New(s.typ)
+	err := dec.DecodeValue(singleResult)
+	return singleResult, err
+}
```

Comment:
This is not called anywhere here, removing all the tests still pass. Can we just remove it from the codebase?

```suggestion
```